services/QuillLMS/app/queries/teacher_dashboard_metrics.rb

```diff
@@ -0,0 +1,54 @@
+class TeacherDashboardMetrics
+
+  def initialize(user)
+    @user = user
+  end
+
+  def run
+   {
+      weekly_assigned_activities_count: count_of_assigned_activities(last_sunday),
+      yearly_assigned_activities_count: count_of_assigned_activities(last_july_first),
+      weekly_completed_activities_count: count_of_completed_activities(last_sunday),
+      yearly_completed_activities_count: count_of_completed_activities(last_july_first)
+    }
```

Comment:
Honest question here: should we be caching this value?  I'm not really sure how expensive these queries might be, but pulling data out of `ActivitySession` has often been pretty time-consuming.  It might make sense to cache this hash by wrapping the whole thing in a `Rails.cache.fetch` block with an expiration of 24 hours and basically only recalculate it once a day.  Not sure how that might impact UX, though, or teacher expectations that their numbers will go up immediately after they assign new stuff.

Reply:
Yeah, I hear that. I think it makes sense to deploy as is and then add caching if necessary, especially because, as you point out, we might have to think about whether to invalidate caches on unit assignment or something.